src/msg/message.cc

```diff
@@ -81,16 +72,52 @@ void CRMessageBase::Unsubscribe(const channel_id_t channel, CRNode* node) {
     }
 }
 
-cr_timestamp_nsec_t CRMessageBase::GetLatestDeliveredTime(const channel_id_t channel) {
+void SubscriptionMap::Dispatch(const CRMessageBasePtr& message) {
+    std::shared_lock lck(mtx_);
```

Comment:
why it's shared?

Reply:
Because we are only reading the subscription list but not modifying them.